lib/vector/Vlib/geos_to_wktb.c

```diff
@@ -266,18 +298,21 @@ char *Vect_line_to_wkt(const struct line_pnts *points, int type, int with_z)
     }
 
     GEOSWKTWriter_setOutputDimension(writer, with_z ? 3 : 2);
+    GEOSWKTWriter_setTrim(writer, trim);
 
     GEOSGeometry *geom = Vect_line_to_geos(points, type, with_z);
 
     if (!geom) {
-        return (NULL);
+        return NULL;
     }
 
     wkt = GEOSWKTWriter_write(writer, geom);
+    char *wkt_out = G_store(wkt);
 
     GEOSGeom_destroy(geom);
+    GEOSFree(wkt);
```

Comment:
What was freeing wkt before? Is it in these conditions that double free issues arise? (I only know the name of the issue, I'd like to learn more to have more efficient reviews)

Reply:
It is with this PR the same as before: it is the callers responsibility to free.
However the [GEOS team recommend](https://libgeos.org/usage/c_api/#readers-and-writers) to free the result of GEOSWKTWriter with `GEOSFree()`. `G_store()` "transfers" that to GRASS (to be free'd with `G_free()`.

Reply:
Just added an improvement of the description of the return values, so it is clear they need to be freed.

Reply:
It's clear they are allocated by the function with your latest commit.